shared/test/test_files.rb

```diff
@@ -344,6 +344,38 @@ def test_bogus_metadata
     assert not_found?
   end
 
+  def test_rename_operations
```

Comment:
Ah, oops, right (testing case and spaces, no overwrite).

Does the API support rename with overwrite? If so, I can add another test in a follow-up PR (alternately, feel welcome to add here).